ci/tasks/configure_libvirt_gpu.py

```diff
@@ -0,0 +1,103 @@
+import argparse
+import logging
+import os
+import subprocess
+import time
+import xml.etree.ElementTree as ET
+
+logger = logging.getLogger('Libvirt-Attach')
+
+
+class ConfigureLibvirtDevice:
+
+    def shutdown_domain(self, domain):
+        if 'shut off' not in self._get_domain_status(domain):
+            logger.info(f'Shutting down {domain}')
+            self._run_cmd(f'virsh shutdown {domain}')
+
+            if not self._wait_for_status(domain, 'shut off'):
+                logger.warning(f'{domain} still not off, forcing off')
+                self._run_cmd(f'virsh destroy {domain}')
+
+    def detach_device(self, device_id):
+        logger.info(f'Detaching {device_id} from host')
+        self._run_cmd(f'virsh nodedev-dettach {device_id}')
+
+    def attach_device(self, domain, device_id):
+        logger.info(f'Attaching {device_id} to {domain}')
+        device_addresses = configure._get_device_addresses(device_id)
+        config_file = configure._write_config_file(f'gpu_{device_id}', device_addresses)
```

Comment:
Is it worth checking here if the list of addresses comes back empty (like if someone mistypes a device id)?